mautic/mautic
# Add "Export to Excel" functionality for contacts

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | staging
| Bug fix?                               | no
| New feature?                           | yes
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | yes
| Related user documentation PR URL      | N/A?
| Related developer documentation PR URL | N/A
| Issue(s) addressed                     | N/A




#### Description:

This PR does two things:

1. Remove dependency on `sonata-project/exporter` since it does basically the same as `phpoffice/phpspreadsheet` (but the latter [has support for more formats](https://phpspreadsheet.readthedocs.io/en/latest/#file-formats-supported), like CSV/Excel/HTML/PDF/etc.). The first was used for exporting contacts while the latter is used for exporting e.g. form results:

![image](https://user-images.githubusercontent.com/17739158/99820572-40890900-2b51-11eb-904c-699c70aaff97.png)

This PR makes that more consistent by using `phpoffice/phpspreadsheet` for exporting contacts as well.

2. Add "Export to Excel" button to the Contacts overview. This is especially handy if you need to export all contacts in a certain segment, or simply all contacts that are in your instance. **This is an enhancement to the already existing export feature for contacts.**




#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. The Contacts page now has two new buttons:

![image](https://user-images.githubusercontent.com/17739158/99820966-b8efca00-2b51-11eb-8c53-d00cbf58dc5d.png)

3. Test both buttons to see if you get valid Excel and CSV files. **Make sure you have at least one contact in your instance**.

**Please let me know if the new buttons are too intrusive: the new buttons can also be put under the down arrow 🔽, where the "Export" button previously was**.
